### DISCUSSION TITLE
PerformIngameSpawn Question.

### DISCUSSION THREAD

**Discussion by brannik** [azerothcore/azerothcore-wotlk#8054] [General]
Hello im using AzerothCore 3.3.5 With Eluna module installed. Currently trying to spawn NPC from lua script. Since the script is triggered by the server not from playeri must use PerformIngameSpawn not SpawnCreature, anyway NPC dont appear at the location buuut when i restart worldserver is on proper location. How i can fix this ? xD

`NPC_SPWNED = PerformIngameSpawn(1,NPC_ID,map,0,x,y,z,o,0,0,1)`
Where all values are set in array ( will spawn NPC on random location) anyway thats not the problem..

**Comment by Helias** [comment]
https://www.azerothcore.org/wiki/how-to-ask-for-help
